SpatialGDK/Source/SpatialGDK/Public/Interop/Connection/ConnectionConfig.h

```diff
@@ -99,15 +99,24 @@ struct FConnectionConfig
 		if (LinkProtocolString.Compare(TEXT("Tcp"), ESearchCase::IgnoreCase) == 0)
 		{
 			LinkProtocol = WORKER_NETWORK_CONNECTION_TYPE_MODULAR_TCP;
+			return;
 		}
 		else if (LinkProtocolString.Compare(TEXT("Kcp"), ESearchCase::IgnoreCase) == 0)
 		{
 			LinkProtocol = WORKER_NETWORK_CONNECTION_TYPE_MODULAR_KCP;
+			return;
 		}
-		else if (!LinkProtocolString.IsEmpty())
+
+		// None given or not recognised, default to KCP for clients, otherwise keep existing TCP default
+		if (SpatialConstants::DefaultClientWorkerType.ToString().Equals(WorkerType))
+		{
+			LinkProtocol = WORKER_NETWORK_CONNECTION_TYPE_MODULAR_KCP;
+		}
+
+		if (!LinkProtocolString.IsEmpty())
 		{
-			UE_LOG(LogTemp, Warning, TEXT("Unknown network protocol %s specified for connecting to SpatialOS. Defaulting to KCP."),
-				   *LinkProtocolString);
+			UE_LOG(LogTemp, Warning, TEXT("Unknown network protocol %s specified for connecting to SpatialOS. Defaulting to %s."),
+				   *LinkProtocolString, LinkProtocol);
```

Comment:
Wait. Does printing the enum `Worker_NetworkConnectionType` as a string work here?

Reply:
Good spot :)